source/common/http/http1/codec_impl.h

```diff
@@ -156,6 +156,8 @@ class ResponseEncoderImpl : public StreamEncoderImpl, public ResponseEncoder {
     return stream_error_on_invalid_http_message_;
   }
 
+  void setRequestDecoder(Http::RequestDecoder& /*decoder*/) override {}
```

Comment:
or maybe why HTTP/1.1 doesn't need this

Reply:
Added. H/1 will need more work to actually reset the decoder on internal redirect, but it seems like we don't need to tackle that now?

Reply:
We will need to make it work for H1 to defer logging I guess? Can you add a TODO?